PWGJE/TableProducer/emcalCorrectionTask.cxx

```diff
@@ -956,28 +1304,46 @@ struct EmcalCorrectionTask {
     trackGlobalIndex.reserve(nTracksInCol);
     fillTrackInfo<decltype(groupedTracks)>(groupedTracks, trackPhi, trackEta, trackGlobalIndex);
 
-    int nClusterInCol = mAnalysisClusters.size();
-    std::vector<float> clusterPhi;
-    std::vector<float> clusterEta;
-    clusterPhi.reserve(nClusterInCol);
-    clusterEta.reserve(nClusterInCol);
+    indexMapPair = matchTracksToCluster(mClusterPhi, mClusterEta, trackPhi, trackEta, maxMatchingDistance, 20);
+  }
 
-    // TODO one loop that could in principle be combined with the other
-    // loop to improve performance
-    for (const auto& cluster : mAnalysisClusters) {
-      // Determine the cluster eta, phi, correcting for the vertex
-      // position.
-      auto pos = cluster.getGlobalPosition();
-      clusterPhi.emplace_back(TVector2::Phi_0_2pi(pos.Phi()));
-      clusterEta.emplace_back(pos.Eta());
+  template <typename Collision>
+  void doSecondaryTrackMatching(Collision const& col, EMV0Legs const& v0legs, MatchResult& indexMapPair, std::vector<int64_t>& trackGlobalIndex, MyGlobTracks const& tracks)
+  {
+    auto groupedV0Legs = v0legs.sliceBy(perCollisionEMV0Legs, col.globalIndex());
+    int nLegsInCol = groupedV0Legs.size();
+    std::vector<float> trackPhi;
+    std::vector<float> trackEta;
+    // reserve memory to reduce on the fly memory allocation
+    trackPhi.reserve(nLegsInCol);
+    trackEta.reserve(nLegsInCol);
+    trackGlobalIndex.reserve(nLegsInCol);
+
+    float trackEtaEmcal = 0.f;
+    float trackPhiEmcal = 0.f;
+    // fillTrackInfo<decltype(groupedTracks)>(groupedTracks, trackPhi, trackEta, trackGlobalIndex);
+    for (const auto& leg : groupedV0Legs) {
+      auto track = tracks.iteratorAt(leg.trackId());
```

Comment:
in case somebody filtered tracks in a different task, i think it is safer to perform some form of check to see if tracks is out of bounds or if the track is found in the track table

Reply:
I am unsure how to perform bound checks on tables properly. Do you know by chance?